lib/easybill-rest-client/api_client.rb

```diff
@@ -26,55 +29,58 @@ def request_collection(method, endpoint, params = {})
     end
 
     def request(method, endpoint, params = {})
-      Retryable.retryable(
-        tries: tries,
-        sleep: retry_cool_off_time,
-        on: EasybillRestClient::TooManyRequests
-      ) do
-        response = perform_request(method, endpoint, params)
-        process_response(response)
+      @request_logger = RequestLogger.new(logger: logger, request_id: Time.now.to_f)
+      retry_on(EasybillRestClient::TooManyRequests) do
+        retry_on(Net::OpenTimeout) do
+          perform_request(method, endpoint, params).body
+        end
       end
+    rescue => e
+      request_logger.error("Request failed due to: #{e.class}: #{e.message}")
+      raise
     end
 
     private
 
-    def perform_request(method, endpoint, params)
-      request_builder = RequestBuilder.new(method)
-      uri = request_builder.build_uri(endpoint, params)
-      request = request_builder.build_request(api_key, uri.request_uri, params)
-      request_details = request_builder.request_details(uri, request)
-      log_request(method, endpoint, request_details)
-
-      Net::HTTP.start(uri.host, uri.port, use_ssl: uri.scheme == 'https') do |http|
-        http.request(request)
+    def retry_on(klass)
+      Retryable.retryable(retryable_opts(klass)) do
+        yield
       end
     end
 
-    def log_request(method, endpoint, params)
-      logger.info("[easybill-rest-client] #{method.to_s.upcase} #{endpoint} #{params}")
+    def retryable_opts(klass)
```

Comment:
Hmm. Have you actually figured out why it can't connect? Are their servers down or just really slow? Or is it maybe something completely different?